### PR TITLE
feat(1881): build cluster to support multple scm conexts

### PR DESCRIPTION
**Pull request by parthasl**
## Context

Build cluster supports only one SCM context, which leads to error "This pipeline is not authorized to use this build cluster." if the same build cluster is used for different SCM context.

## Objective

This PR is intended for a build cluster to support multiple SCM contexts

## References

[1881](https://github.com/screwdriver-cd/screwdriver/issues/1881)

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: test/data/buildCluster.create.yaml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # BuildCluster create example
 name: iOS
+scmContexts: ['github:github.com', 'github:git.com']
```
Comment (DekusDenial) [inline test/data/buildCluster.create.yaml]: Is `git.com` a typo?

File: migrations/20191214-rename_scmContext-buildClusters.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+'use strict';
+
+const prefix = process.env.DATASTORE_SEQUELIZE_PREFIX || '';
+const table = `${prefix}buildClusters`;
+
+module.exports = {
+    up: async (queryInterface, Sequelize) => {
+        try {
+            await queryInterface.sequelize.query('SET SQL_MODE = CONCAT(@@SQL_MODE, ' +
+                '\',ANSI_QUOTES\')');
```
Comment (DekusDenial) [inline migrations/20191214-rename_scmContext-buildClusters.js]: Will Postgres ignore these?

File: test/data/buildCluster.create.yaml (hunk 1 above)
Comment (parthasl) [reply test/data/buildCluster.create.yaml]: no, just gave some random scm context

File: migrations/20191214-rename_scmContext-buildClusters.js (hunk 2 above)
Comment (parthasl) [reply migrations/20191214-rename_scmContext-buildClusters.js]: yes, it will ignore

File: test/data/buildCluster.create.yaml (hunk 1 above)
Comment (parthasl) [reply test/data/buildCluster.create.yaml]: updated

**Pull request merged**